### PR TITLE
Add support for <https://m2mobi.com> style links

### PR DESCRIPTION
**Pull request by jvanzummeren**
As filed by SMiller at https://github.com/M2mobi/Marky-Mark/issues/110.

This Pull request addd support for <https://m2mobi.com> style links

### CODE REVIEW

**Review by SMillerDev — APPROVED** [review]
Tests look good to me, I can't say much about the actual implementation though.

File: markymark/Classes/Rules/Inline/ShortLinkRule.swift [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    }
+
+    // MARK: Rule
+
+    public func recognizesLines(_ lines: [String]) -> Bool {
+        guard let line = lines.first else { return false }
+        return getResults(line).count > 0
+    }
+
+    public func createMarkDownItemWithLines(_ lines: [String]) -> MarkDownItem {
```
Comment (Basca) [inline markymark/Classes/Rules/Inline/ShortLinkRule.swift:30]: Sometimes these are open and something they aren't for all the rules.
Was there a specific reasoning for that?

File: markymark/Classes/Rules/Inline/ShortLinkRule.swift [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+            lines: lines,
+            content: url ?? "",
+            url: url ?? ""
+        )
+    }
+}
+
+private extension ShortLinkRule {
+
+    private func getResults(_ string: String) -> [NSTextCheckingResult] {
```
Comment (Basca) [inline markymark/Classes/Rules/Inline/ShortLinkRule.swift:43]: I thought xcode wouldn't say these private annotations are redundant as the extension itself is already private.

File: markymark/Classes/Rules/Inline/ShortLinkRule.swift [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+        
+        let validResults = results.filter {
+            isValidURLString(string.subString($0.range(at: 1)))
+        }
+        
+        return validResults
+    }
+    
+    private func isValidURLString(_ string: String) -> Bool {
+        let detector = try! NSDataDetector(types: NSTextCheckingResult.CheckingType.link.rawValue)
```
Comment (Basca) [inline markymark/Classes/Rules/Inline/ShortLinkRule.swift:55]: do we really want to try! here? as there is no catch

File: Example/Tests/Rules/Inline/ShortLinkRuleTests.swift [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+        XCTAssertTrue(sut.recognizesLines(["<https://www.m2mobi.com>"]))
+        XCTAssertTrue(sut.recognizesLines(["<http://www.m2mobi.nl/something/other?id=4&page=4>"]))
+        XCTAssertTrue(sut.recognizesLines(["<hkia://flights>"]))
+        XCTAssertTrue(sut.recognizesLines(["<google.com>"]))
+    }
+
+    func test_WillNotRecognizesLines_When_CallingRecognizesLinesWithInvalidInput() {
+        XCTAssertFalse(sut.recognizesLines(["<Google>"]))
+        XCTAssertFalse(sut.recognizesLines(["<$$>"]))
+        XCTAssertFalse(sut.recognizesLines(["<hello>"]))
```
Comment (Basca) [inline Example/Tests/Rules/Inline/ShortLinkRuleTests.swift:31]: is it possible to test for "<>" ? I believe the regex also supports empty elements for this rule.

**Review by Basca — CHANGES_REQUESTED** [review]

File: markymark/Classes/Rules/Inline/ShortLinkRule.swift (hunk 1 above)
Comment (jvanzummeren) [reply markymark/Classes/Rules/Inline/ShortLinkRule.swift:30]: What exactly do you mean? The `open` accessor is only available for classes and determines wether a class can be subclassed or not in comparisson to `public`. For function's there is nothing more public than public

File: markymark/Classes/Rules/Inline/ShortLinkRule.swift (hunk 2 above)
Comment (jvanzummeren) [reply markymark/Classes/Rules/Inline/ShortLinkRule.swift:43]: Solved

**Review by Basca — APPROVED** [review]

**Pull request merged**